contrib/views/person.py

```diff
@@ -0,0 +1,190 @@
+# ##############################################################################
+#
+#    OSIS stands for Open Student Information System. It's an application
+#    designed to manage the core business of higher education institutions,
+#    such as universities, faculties, institutes and professional schools.
+#    The core business involves the administration of students, teachers,
+#    courses, programs and so on.
+#
+#    Copyright (C) 2015-2021 Université catholique de Louvain (http://www.uclouvain.be)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    A copy of this license - GNU General Public License - is available
+#    at the root of the source code of this program.  If not,
+#    see http://www.gnu.org/licenses/.
+#
+# ##############################################################################
+from django.conf import settings
+from django.urls import reverse_lazy
+from django.utils.translation import get_language
+from django.views.generic import FormView, TemplateView
+
+from admission.services.autocomplete import AdmissionAutocompleteService
+from osis_document.api.utils import get_remote_token
+
+from admission.contrib.forms.person import (
+    DoctorateAdmissionAddressForm,
+    DoctorateAdmissionCoordonneesForm,
+    DoctorateAdmissionPersonForm,
+    get_countries_choices,
+)
+from admission.services.mixins import ApiExceptionErrorMappingMixin
+from admission.services.person import AdmissionPersonService
+
+__all__ = [
+    "DoctorateAdmissionPersonFormView",
+    "DoctorateAdmissionPersonDetailView",
+    "DoctorateAdmissionCoordonneesFormView",
+    "DoctorateAdmissionCoordonneesDetailView",
+]
+
+
+class DoctorateAdmissionCoordonneesDetailView(TemplateView):
+    template_name = 'admission/doctorate/detail_coordonnees.html'
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        coordonnees = AdmissionPersonService.retrieve_person_coordonnees()
+        context_data['coordonnees'] = coordonnees
+        context_data['countries'] = {
+            c.pk: (c.name if get_language() == settings.LANGUAGE_CODE else c.name_en)
+            for c in AdmissionAutocompleteService().autocomplete_countries()
+        }
+        # check if there is at least one data into contact
+        for k in coordonnees["contact"].attribute_map:
+            context_data["show_contact"] = True if coordonnees["contact"][k] else False
```

Comment:
C'est potentiellement un bug si le dernier champ de la boucle est vide:

Préférer `context_data["show_contact"] = any(coordonnees["contact"][k] for k in coordonnees["contact"].attribute_map)`